File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int right = 1;
        int max = 0;
        while(right < prices.length){
            if(prices[right]-prices[left] <= 0){
                left = right;
                right++;
            }else{
                max = Math.max(max, prices[right]-prices[left]);
                right++;
            } 
           
        }
        
        return max;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        currBuy = 0
        currSell = 0
        profit = 0
        for i in range(0,len(prices)):
            if prices[i] < prices[currBuy]:
                profit = max(profit, prices[currSell]-prices[currBuy])
                currBuy = i
                currSell = i
            else:
                if prices[i] > prices[currSell]:
                    currSell = i
                    profit = max(profit, prices[currSell]-prices[currBuy])
        
        return profit